lib/streamlit/elements/camera_input.py

```diff
@@ -56,15 +57,24 @@ def _get_upload_files(
     if len(uploaded_file_info) == 0:
         return []
 
-    active_files = {f.file_id: f.file_urls for f in uploaded_file_info}
-
-    file_recs = ctx.uploaded_file_mgr.get_files(
+    file_recs_list = ctx.uploaded_file_mgr.get_files(
         session_id=ctx.session_id,
-        file_ids=list(active_files.keys()),
+        file_ids=[f.file_id for f in uploaded_file_info],
     )
-    return [
-        UploadedFile(file_rec, active_files[file_rec.file_id]) for file_rec in file_recs
-    ]
+
+    file_recs = {f.file_id: f for f in file_recs_list}
+
+    collected_files: List[Union[UploadedFile, DeletedFile]] = []
+
+    for f in uploaded_file_info:
+        maybe_file_rec = file_recs.get(f.file_id)
+        if maybe_file_rec is not None:
```

Comment:
two nits:
* `maybe_file_rec` doesn't seem to be used, so could we drop the intermediary variable and instead write `if f.file_id in file_recs`?
* this code is more or less duplicated in `file_uploader.py` so it'd probably be worth factoring it out into a helper function (maybe all of `_get_upload_files` can be duplicated, but I'm not sure)